### DISCUSSION TITLE
Migration from v1.5.1 to v2.0.0 failed

### DISCUSSION THREAD

**Discussion by 0xgeorgiev** [plausible/analytics#3765] [Self-Hosted Support]
I recently tried to perform update to my Plausible application from version 1.5.1 to version 2.0.0. I updated the image version and restarted the container. The application then started without issues but didn't show any data as expected due to the new tables in Clickhouse database. I ran the data migration script and it successfully created the new tables, but failed when moving data from the old ones to the new. This is the error message:
`** (MatchError) no match of right hand side value: {:error, %Ch.Error{code: 424, message: "Code: 424. DB::errnoexception: Cannot link /var/lib/clickhouse/store/e25/e25b9758-d799-4d37-ae8e-1cb7b36ee0e4/202306_1_1_0/checksums.txt to /var/lib/clickhouse/store/8cb/8cb0cf28-73cb-4aea-9971-7101ed036166/tmp_replace_from_202306_10_10_0/checksums.txt, errno: 524, strerror: Unknown error 524. (CANNOT_LINK) (version 23.4.2.11 (official build))\n"}}
    lib/plausible/data_migration/numeric_ids.ex:6: Plausible.DataMigration.NumericIDs.do_run/2
    lib/plausible/data_migration/numeric_ids.ex:145: anonymous fn/1 in Plausible.DataMigration.NumericIDs.run/1
    lib/plausible/data_migration/numeric_ids.ex:143: anonymous fn/3 in Plausible.DataMigration.NumericIDs.run/1
    (elixir 1.14.3) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/plausible/data_migration/numeric_ids.ex:140: anonymous fn/2 in Plausible.DataMigration.NumericIDs.run/1
    nofile:1: (file)
    (stdlib 4.2) erl_eval.erl:748: :erl_eval.do_apply/7
command terminated with exit code 1`
It seems like a permission issue, but I still cannot find the reason why it fails. Any help would be appreciated!

I am using Clickhouse version 23.4.2.11-alpine.

**Comment by saranshisatgit** [comment]
Just a word of advise the docker compose file contains a migration script that might not be really useful in case when migrating services, so maybe remove that though its optional
